go-controller/pkg/ovn/egressip.go

```diff
@@ -1843,9 +1843,10 @@ func (e *egressIPController) addPerPodGRSNAT(podNamespace, podName string, podIP
 	return nil
 }
 
-func (e *egressIPController) deletePerPodGRSNAT(pod *kapi.Pod, podIPs []*net.IPNet) error {
-	if config.Gateway.DisableSNATMultipleGWs {
-		// remove snats to->nodeIP (from the node where pod exists) for these podIPs before adding the snat to->egressIP
+func (e *egressIPController) deletePerPodGRSNAT(pod *kapi.Pod, podIPs []*net.IPNet, status egressipv1.EgressIPStatusItem) error {
+	if config.Gateway.DisableSNATMultipleGWs && status.Node == pod.Spec.NodeName {
```

Comment:
@tssurya do you think it would be good to have an info log here to indicate we did not delete because 
`status.Node != pod.Spec.NodeName` ?

```
if config.Gateway.DisableSNATMultipleGWs {
   if status.Node != pod.Spec.NodeName {
       log ( "will not deletePerPodGRSNAT because pod is in the egress node ans Surya fixed this now!" )
   } else {
     ...
```

Reply:
good point, might create some noise though maybe log5...